Automattic/wp-calypso
# Email Comparison: UI improvements to display introductory offers and discounts more accurately

To support the regional pricing work we are currently wrangling in the backend, we need a few pieces in the frontend to ensure that we display discounts and product offers properly.

We now also have a situation where Google Workspace for instance has a few currencies with diverse billing plans and introductory offers with discounts spanning over different periods. We have to represent this more flexibly in the pricing display.

Sometimes the target product also has a regular sale coupon attached and we have to factor that into the display as well.

#### Proposed Changes

* Extend the introductory product offer display to support non-zero discounts. We used to only display zero pricing, optimized for free trials.
* Extend the offer eligibility for email products to include new domains in the cart. 
* Extend the display of offers to dynamically include the interval unit and count.
* Factor in renewal discounts over periods i.e. `transition_after_renewal_count`
* Rename unit exporting symbol `isDomainEligibleForTitanFreeTrial()` to `isDomainEligibleForTitanIntroductoryOffer()` as it more aptly defines its functionality since we need to make a distinction between a domain that is eligible for an introductory offer, and one that is additionally eligible for a free trial ( i.e. zero cost )

#### Testing Instructions



* Apply D85347-code to the backend. Also apply the test instruction to set up your environment. That patch provides simulated billing plans for certain currencies to make it easier to test various introductory offer and sale coupon scenarios.
* Load the stacked email comparison view in your local dev server i.e. http://calypso.localhost:3000/email/[:site] . You can also use [Calypso Live](https://github.com/Automattic/wp-calypso/pull/66153#issuecomment-1201968909) to navigate that view. You can also reach this view by attempting to purchase a new domain, which leads you to the email upsell. Either way, always ensure to use a domain that has not been used to purchase email before.
* The best way to test the price information unit is to have about four users, with four different currencies: 
    * MXN - free trials on Professional Email, discounted introductory offers on Google Workspace
    * PHP - free trials on Google Workspace, discounted introductory offers on Professional Email
    * TRY - no introductory offer
    * USD - default system billing plans
    * You can also mix in a sale coupon


<img width="800" alt="Screenshot 2022-08-14 at 10 16 41 PM" src="https://user-images.githubusercontent.com/277661/184555281-dab85a3d-4e26-4da5-96cd-1dd71e3673ba.png">

<img width="800" alt="Screenshot 2022-08-14 at 10 16 51 PM" src="https://user-images.githubusercontent.com/277661/184555286-541f7233-8746-4ceb-b657-dd8cb2c37779.png">

<img width="800" alt="Screenshot 2022-08-14 at 10 17 04 PM" src="https://user-images.githubusercontent.com/277661/184555296-76d12ba7-116c-436d-9697-0f81a4d6583a.png">

<img width="800" alt="Screenshot 2022-08-14 at 10 17 12 PM" src="https://user-images.githubusercontent.com/277661/184555304-cdd2caba-039b-45d3-b3bc-df87bcc34b08.png">

<img width="800" alt="Screenshot 2022-08-14 at 10 17 33 PM" src="https://user-images.githubusercontent.com/277661/184555307-e9b1f399-88e0-420c-8add-b0f40424e27f.png">

<img width="800" alt="Screenshot 2022-08-14 at 10 17 42 PM" src="https://user-images.githubusercontent.com/277661/184555309-c458334c-99ff-4261-ac38-ff215eb2513d.png">

<img width="800" alt="Screenshot 2022-08-14 at 10 18 06 PM" src="https://user-images.githubusercontent.com/277661/184555315-df228136-dcaf-4969-848a-a6a6d27bcea0.png">

<img width="800" alt="Screenshot 2022-08-14 at 10 18 18 PM" src="https://user-images.githubusercontent.com/277661/184555319-57b9fd4e-deef-4f17-a13f-c5b21596277d.png">




#### Pre-merge Checklist
